tests/pipeline_backend_test.py

```diff
@@ -588,7 +593,7 @@ def assert_laziness(operator, *args):
         assert_laziness(self.backend.sum_per_key)
         assert_laziness(self.backend.flat_map, str)
         assert_laziness(self.backend.sample_fixed_per_key, int)
-        assert_laziness(self.backend.filter_by_key, list)
```

Comment:
why so?

Reply:
we need somthing that it's converted to set.